ros/processor/suggestions_engine.py

```diff
@@ -22,6 +28,141 @@ def __init__(self):
         self.service = 'SUGGESTIONS_ENGINE'
         self.event = None
 
+    def run_pmlogextract(self, host, index_file_path, output_dir):
+        """Run the pmlogextract command."""
+
+        pmlogextract_command = [
+            "pmlogextract",
+            "-c",
+            "ros/lib/pcp_extract_config",
+            index_file_path,
+            output_dir
+        ]
+
+        logging.debug(f"{self.service} - {self.event} - Running pmlogextract command for system {host.get('id')}.")
+        try:
+            subprocess.run(pmlogextract_command, check=True)
+            logging.debug(
+                f"{self.service} - {self.event} - Successfully ran pmlogextract command for system {host.get('id')}."
+            )
+        except subprocess.CalledProcessError as error:
+            logging.error(
+                f"{self.service} - {self.event} - Error running pmlogextract command for system {host.get('id')}:"
+                f" {error.stdout}"
+            )
+
+    def run_pmlogsummary(self, host, output_dir):
+        """Run the pmlogsummary command."""
+
+        pmlogsummary_command = [
+            "pmlogsummary",
+            "-F",
+            os.path.join(output_dir, ".index")
+        ]
+
+        logging.debug(f"{self.service} - {self.event} - Running pmlogsummary command for system {host.get('id')}.")
+        try:
+            subprocess.run(pmlogsummary_command, check=True, text=True, capture_output=True)
+            logging.debug(
+                f"{self.service} - {self.event} - Successfully ran pmlogsummary command for system {host.get('id')}."
+            )
+        except subprocess.CalledProcessError as error:
+            logging.error(
+                f"{self.service} - {self.event} - Error running pmlogsummary command for system {host.get('id')}:"
+                f" {error.stdout}"
+            )
+
+    def create_output_dir(self, request_id, host):
+        output_dir = f"/var/tmp/pmlogextract-output-{request_id}/"
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+
+        logging.debug(f"Successfully created output_dir for system {host.get('id')}: {output_dir}")
+
+        return output_dir
+
+    def run_pcp_commands(self, host, index_file_path, request_id):
+        sanitized_request_id = request_id.replace("/", "_")
+        output_dir = self.create_output_dir(sanitized_request_id, host)
+
+        self.run_pmlogextract(host, index_file_path, output_dir)
+
+        self.run_pmlogsummary(host, output_dir)
+
+        logging.info(f"{self.service} - {self.event} - Successfully ran pcp commands for system {host.get('id')}.")
```

Comment:
I think your previous approach was correct, as I was in doubt I ended up doing this locally and it shows, if `run_pmlogextract` fails still we end up processing the `run_pmlogsummary`. Though `run_pmlogsummary` will throw exception as well however we can avoid that. So can we chain the exceptions again?